test/language/expressions/object/computed-property-evaluation-order.js

```diff
@@ -0,0 +1,26 @@
+// Copyright (C) 2016 Michael Ficarra. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-object-initializer-runtime-semantics-propertydefinitionevaluation
+description: >
+    Evaluation of PropertyDefinitionList occurs in order, and each
+    PropertyDefinition's PropertyName is evaluated before its
+    AssignmentExpression.
+---*/
+
+var counter = 0;
+var o = {
+  [++counter]: ++counter,
+  [++counter]: ++counter,
+  [++counter]: ++counter,
+}
+
+var keys = Object.keys(o).sort();
```

Comment:
I wanted it to work in an implementation that doesn't necessarily implement `Object.keys` ordering properly. This is not an `Object.keys` test.